core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1206,4 +1206,61 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return ExprEval.of(expr.value() != null, ExprType.LONG);
     }
   }
+
+  class LpadFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "lpad";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 3) {
+        throw new IAE("Function[%s] needs 3 arguments", name());
+      }
+      
+      String base = NullHandling.nullToEmptyIfNeeded(args.get(0).eval(bindings).asString());
```

Comment:
Does it make sense to pad empty values when sql compatible null handling is not enabled?